### PR TITLE
Feat(web): Introduce JavaScript handler for Password toggle (refs

### PR DESCRIPTION
**Pull request by literat**
Introducing Password Field toggle JS handler to show and hide passwords.

### CODE REVIEW

**Review by crishpeen — APPROVED** [review]

File: packages/web/src/js/Password.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+function handlePasswordClick() {
+  const target = getElement(this);
+
+  if (target) {
+    const password = new Password(target);
+    password.toggle(target);
+  }
+}
+
+// When document content is loaded
```
Comment (adamkudrna) [inline packages/web/src/js/Password.ts:50]: Do you think the comments are necessary…? 🤔 I believe the code itself is readable enough (+1 for that). The same goes for `Header` handler.

File: packages/web/src/js/Password.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    target.setAttribute(PASSWORD_ARIA_LABEL, 'Hide password');
+    SelectorEngine.findOne(PASSWORD_FIELD_ELEMENT, target.parentElement as HTMLElement).setAttribute('type', 'text');
+
+    this.isShown = true;
+  }
+
+  hide(target: HTMLElement) {
+    target.setAttribute(PASSWORD_ARIA_PRESSED, 'false');
+    target.setAttribute(PASSWORD_ARIA_LABEL, 'Show password');
+    SelectorEngine.findOne(PASSWORD_FIELD_ELEMENT, target.parentElement as HTMLElement).setAttribute('type', 'password');
```
Comment (adamkudrna) [inline packages/web/src/js/Password.ts:31]: GH check should be fixed.

**Review by adamkudrna — CHANGES_REQUESTED** [review]
Does not work on my machine, sorry! (FF/macOS) 🤷🏻 Password gets revealed, but I cannot hide it once its shown.

File: packages/web/src/js/Password.ts (hunk 1 above)
Comment (literat) [reply packages/web/src/js/Password.ts]: Ok, this was fine during step-by-step implementation for me to better grab that logic. Now is readable enough. I agree. Removing them.

**Comment by literat** [comment]
> Does not work on my machine, sorry! (FF/macOS) 🤷🏻 Password gets revealed, but I cannot hide it once its shown.

I was missing multiple instances on page in first iteration. Now it is fixed. Even in FF ;-)

File: packages/web/src/js/Password.ts (hunk 2 above)
Comment (literat) [reply packages/web/src/js/Password.ts]: Fixed.

**Review by adamkudrna — APPROVED** [review]
So many lines of code… but it works now! 🎉 

obrazek

**Comment by adamkudrna** [comment]
> I was missing multiple instances on page in first iteration. Now it is fixed. Even in FF ;-)

Could testing multiple instances be useful? Or is it not necessary now when we have `InstanceMap`?

**Comment by literat** [comment]
> Could testing multiple instances be useful? Or is it not necessary now when we have `InstanceMap`?

I've also added some tests to BaseComponent to test this feature. And yes, we have also tests on InstanceMap where multiple instances are tested. So I think this is not necessary to duplicate test cases.

**Comment by literat** [comment]
> So many lines of code… but it works now! 🎉
> 
> obrazek

I know but this will be helpful in near future when will be dealing with multiple modals in one page.

**Pull request merged**